chain/client/src/sync/epoch.rs

```diff
@@ -483,15 +472,20 @@ impl EpochSync {
             proof.last_epoch.next_next_epoch_info,
         )?;
 
+        let last_epoch_data = &proof.past_epochs[proof.past_epochs.len() - 2];
+        let last_epoch_last_final_block_height = last_epoch_data.last_final_block_header.height();
+        let first_block_info_in_epoch =
+            BlockInfo::from_header(&last_header, last_epoch_last_final_block_height);
+
         store_update.insert_ser(
             DBCol::BlockInfo,
-            &borsh::to_vec(proof.current_epoch.first_block_info_in_epoch.hash()).unwrap(),
-            &proof.current_epoch.first_block_info_in_epoch,
+            &borsh::to_vec(first_block_info_in_epoch.hash()).unwrap(),
+            &first_block_info_in_epoch,
         )?;
 
         store_update.set_ser(
             DBCol::BlockOrdinal,
-            &borsh::to_vec(&proof.current_epoch.merkle_proof_for_first_block.size()).unwrap(),
+            &index_to_bytes(last_header.block_ordinal()),
```

Comment:
Concern: `BlockHeader::block_ordinal()` is supported since `BlockHeaderV3`.

Reply:
I don't think that's an issue. EpochSync to a block header version below V3 is unsupportable due to also not having epoch_data_hash.

Reply:
But it will be worth making a test to check that trying to sync to a version below V3 will fail. I'll file an issue.